aptos-move/framework/aptos-framework/sources/object_code_deployment.move

```diff
@@ -0,0 +1,73 @@
+module aptos_framework::object_code_deployment {
+    use std::bcs;
+    use std::features;
+    use std::signer;
+    use std::string;
+    use std::vector;
+    use aptos_framework::account;
+    use aptos_framework::code;
+    use aptos_framework::code::PackageRegistry;
+    use aptos_framework::object;
+    use aptos_framework::object::{ExtendRef, Object};
+
+    /// Object code deployment not supported.
+    const EOBJECT_CODE_DEPLOYMENT_NOT_SUPPORTED: u64 = 0x1;
+    /// Not the owner of the `PublisherRef`
+    const ENOT_OWNER: u64 = 0x2;
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct PublisherRef has key {
+        extend_ref: ExtendRef,
+    }
+
+    /// Create a new object to host the code, and `PublisherRef` if the code is upgradeable,
+    /// Send `PublisherRef` to object signer.
+    public entry fun publish(
+        publisher: &signer,
+        metadata_serialized: vector<u8>,
+        code: vector<vector<u8>>,
+    ) {
+        assert!(features::is_object_code_deployment_enabled(), EOBJECT_CODE_DEPLOYMENT_NOT_SUPPORTED);
+
+        let object_seed = object_seed(signer::address_of(publisher));
+        let constructor_ref = &object::create_named_object(publisher, object_seed);
+        let module_signer = &object::generate_signer(constructor_ref);
+        code::publish_package_txn(module_signer, metadata_serialized, code);
+
+        if (code::is_package_upgradeable(metadata_serialized)) {
+            move_to(module_signer, PublisherRef {
+                extend_ref: object::generate_extend_ref(constructor_ref)
+            });
+        };
+    }
+
+    inline fun object_seed(publisher: address): vector<u8> {
+        let object_seed = &mut string::utf8(b"aptos_framework::object_code_deployment");
+        let sequence_number = account::get_sequence_number(publisher) + 1;
```

Comment:
Should this be current seq number or + 1? What are your thoughts here?

Reply:
I would say current seq num but does it actually matter?

Reply:
My reasoning for `seq_num +1` was because the transaction that will include this deployment will have the user's account seq number incremented, therefore having the same seq number aligned with the corresponding transaction makes sense.